kore/src/Kore/Unparser.hs

```diff
@@ -227,7 +227,7 @@ layoutPrettyUnbounded =
 
  -}
 escapeString :: String -> String
-escapeString s = foldr (.) id (map escapeCharS s) ""
+escapeString s = foldr ((.) . escapeCharS) id s ""
```

Comment:
What's `ala`?

Reply:
It's `Function.on` on steroids from `Control.Lens.Combinators`
```
λ :t ala
ala
  :: (Functor f, Rewrapped s t, Rewrapped t s) =>
     (Unwrapped s -> s)
     -> ((Unwrapped t -> t) -> f s) -> f (Unwrapped s)
```

So you can do stuff like:
```
λ ala Sum foldMap [1, 2, 3 :: Int]
6
λ ala All foldMap [True, False]
False
λ ala Alt foldMap ["Hello, ", "world!"]
"Hello, world!"
λ ala Ap foldMap [Just "1", Just "2"]
Just "12"
λ ala Ap foldMap [Just "1", Nothing]
Nothing
```
Note there's no need to deconstruct the newtypes, `ala` does that for us.

Reply:
Actually, this is the same as `flip (foldr escapeCharS) s`

Reply:
`ala Endo foldMap (escapeCharS <$> s)` is equivalent to the original version, but not the new one. HLint isn't just playing code golf; it is actually more efficient (*) to apply `escapeCharS` during the fold than during a map prior to the fold.

`flip (foldr escapeCharS) s` is the same, or `foldr escapeCharS "" s`.

(*) When optimization is on, GHC performs fusion so that these alternatives are actually the same. Though presumably we would also like reasonable efficiency in GHCi or with profiling.

Reply:
Fair enough. I don't feel strongly about this either way, but I was in the mindset of "these changes make the code easier to read", and this change stands out as a counter-example.